File: The Odin Project/javascript-exercises/calculator/calculator.js
function add(a, b) {
  return a + b;
}

function subtract(a, b) {
  return a - b;
}

function sum(numbers) {
  const total = numbers.reduce(
    (currentValue, number) => currentValue + number,
    0
  );

  return total;
}

function multiply(numbers) {
  const total = numbers.reduce(
    (currentValue, number) => currentValue * number,
    1
  );

  return total;
}

function power(number, power) {
  return Math.pow(number, power);
}

function factorial(number) {
  let res = 1;

  if (number > 0) {
    for (let i = number; i >= 1; i--) {
      res *= i;
    }
  }

  return res;
}

module.exports = {
  add,
  subtract,
  sum,
  multiply,
  power,
  factorial,
};

File: The Odin Project/2-ruby-programming/ruby-testing/calculator.rb
class Calculator
  def add (*numbers)
    sum = numbers.reduce (0) { |total, number| number + total }
    
    sum
  end

  def subtract (initial_number, *numbers)
    sum = numbers.reduce (initial_number) { |total, number| total - number }
    
    sum
  end

  def multiply (*numbers)
    sum = numbers.reduce (1) { |total, number| number * total }
    
    sum
  end

  def divide (initial_number, *numbers)
    sum = numbers.reduce (initial_number) { |total, number| total / number }
    
    sum
  end
end